lib/widgets/content.dart

```diff
@@ -967,9 +970,22 @@ class AvatarImage extends StatelessWidget {
       null          => null, // TODO(#255): handle computing gravatars
       var avatarUrl => store.tryResolveUrl(avatarUrl),
     };
-    return (resolvedUrl == null)
-      ? const SizedBox.shrink()
-      : RealmContentNetworkImage(resolvedUrl, filterQuality: FilterQuality.medium, fit: BoxFit.cover);
+
+    if(resolvedUrl == null) {
+      return const SizedBox.shrink();
+    } else{
+      final avatarUrl = AvatarUrl.fromUserData(
+        rawAvatarUrl: resolvedUrl.toString(),
```

Comment:
If we already have a parsed `Uri` object, we should be passing that instead of a string.

That will also let us simplify some of the downstream logic within AvatarUrl and its subclasses. This `Uri` object is always an absolute URL (because `tryResolveUrl` always returns an absolute URL), so we don't need any of the logic specific to relative URLs.

Reply:
https://github.com/zulip/zulip-flutter/pull/534/commits/4c24c3e6c25f12b792f947839baa8e85c5305e91